modules/programs/fastfetch.nix

```diff
@@ -0,0 +1,96 @@
+{ pkgs, lib, config, ... }:
+
+let
+
+  cfg = config.programs.fastfetch;
+
+  jsonFormat = pkgs.formats.json { };
+
+in {
+  meta.maintainers = [ lib.hm.maintainers.afresquet ];
+
+  options.programs.fastfetch = {
+    enable = lib.mkEnableOption "Fastfetch";
+
+    package = lib.mkPackageOption pkgs "fastfetch" { };
+
+    settings = lib.mkOption {
+      type = jsonFormat.type;
+      default = {
```

Comment:
I assume fastfetch uses the default values when no configuration file is created? If so then perhaps it is possible to have `default = {}` and do

``` nix
    xdg.configFile."fastfetch/config.jsonc" = mkIf (cfg.settings != {}) {
      source = jsonFormat.generate "config.jsonc" cfg.settings;
    };
```

It would help avoid problems when the upstream default configuration changes.